### DISCUSSION TITLE
How to add custom field in session table

### DISCUSSION THREAD

**Discussion by rizkifadilah12** [vendurehq/vendure#1919] [Q&A]
Hello, i want to add 1 field "deviceId" to table session for flagging when user login in web desktop or mobile device, any idea?

**Comment by michaelbromley** [comment]
Hi,

So we don't currently support custom fields on the Session entity, but a possible way to implement this could be to use the [EntityMetadataModifier](https://www.vendure.io/docs/typescript-api/configuration/entity-options/#entitymetadatamodifier) API to add a column to the session table.

**Reply by rizkifadilah12** [reply]
But its possible to edit data token in table session, i want to add separator in field token +_mobile when login in device mobile?
